### PR TITLE
KAFKA-4404: Add javadocs to document core Connect types, especially that integer types are signed

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@hachikuji @ijuma @kkonstantine @shikhar Trivial javadoc update to try to head off further questions around the restriction to only having signed types.

File: connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java [hunk 1]
```diff
@@ -44,7 +44,66 @@
...
+        /**
+         *  8-bit signed integer
+         *
+         *  Note that if you have an unsigned 8-bit data source, {@link Type#INT16} will be required to safely capture all valid values
+         */
+        INT8,
+        /**
+         *  16-bit signed integer
+         *
+         *  Note that if you have an unsigned 8-bit data source, {@link Type#INT32} will be required to safely capture all valid values
```
Comment (kkonstantine) [inline connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java]: if you have an unsigned ~~8-bit~~ 16-bit data source

File: connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java [hunk 2]
```diff
@@ -44,7 +44,66 @@
...
+         */
+        INT16,
+        /**
+         *  32-bit signed integer
+         *
+         *  Note that if you have an unsigned 8-bit data source, {@link Type#INT64} will be required to safely capture all valid values
```
Comment (kkonstantine) [inline connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java]: if you have an unsigned ~~8-bit~~ 32-bit data source

File: connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java [hunk 3]
```diff
@@ -44,7 +44,66 @@
...
+         */
+        INT32,
+        /**
+         *  64-bit signed integer
+         *
+         *  Note that if you have an unsigned 8-bit data source, the {@link Decimal} logical type (encoded as {@link Type#BYTES})
```
Comment (kkonstantine) [inline connect/api/src/main/java/org/apache/kafka/connect/data/Schema.java]: if you have an unsigned ~~8-bit~~ 64-bit data source

**Review by kkonstantine — APPROVED** [review]
Besides a few nits, LGTM by me.

**Comment by hachikuji** [comment]
LGTM

**Pull request closed without merging**